signer/tests/integration/complete_deposit.rs

```diff
@@ -0,0 +1,673 @@
+    let (mut complete_deposit_tx, ctx) = make_complete_deposit(&deposit_req, &sweep_tx, &chain_tip);
+    // Different: Okay, let's make sure we get the deployers do not match.
+    complete_deposit_tx.recipient = fake::Faker
+        .fake_with_rng::<StacksPrincipal, _>(&mut rng)
+        .into();
+
+    let validate_future = complete_deposit_tx.validate(&db, &ctx);
+    match validate_future.await.unwrap_err() {
+        Error::DepositValidation(ref err) => {
+            assert_eq!(err.error, DepositErrorMsg::RecipientMismatch)
+        }
+        err => panic!("unexpected error during validation {err}"),
+    }
+
+    testing::storage::drop_db(db).await;
+}
+
+/// For this test we check that the `CompleteDepositV1::validate` function
+/// returns a deposit validation error with a InvalidMintAmount message
+/// when the amount of sBTC to mint exceeds the amount in the signer's
+/// deposit request record.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn complete_deposit_validation_invalid_mint_amount() {
+    // Normal: this generates the blockchain as well as deposit request
+    // transactions in each bitcoin block.
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let db = testing::storage::new_test_database(db_num, true).await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let signatures_required = 2;
+
+    let signer_set = deposit_setup(&mut rng, &db).await;
+
+    // Get the chain tip.
+    let chain_tip = get_bitcoin_canonical_chain_tip_block(&db).await;
+    // Normal: Get an existing deposit request on the canonical bitcoin
+    // blockchain.
+    let deposit_req =
+        get_pending_accepted_deposit_requests(&db, &chain_tip, signatures_required).await;
+
+    // Normal: we generate a transaction that sweeps in the deposit.
+    let sweep_config = SweepTxConfig {
+        signer_public_key: PublicKey::combine_keys(&signer_set).unwrap(),
+        amounts: 3000..1_000_000_000,
+        inputs: vec![deposit_req.outpoint()],
+    };
+    let mut sweep_tx: model::Transaction = sweep_config.fake_with_rng(&mut rng);
+    // Normal: make sure the sweep transaction is on the canonical bitcoin
+    // blockchain and is in our database.
+    sweep_tx.block_hash = chain_tip.block_hash.into_bytes();
+    // Normal: make sure that we have a record of the sweep transaction in
+    // our database.
+    let bitcoin_tx_ref = BitcoinTxRef {
+        txid: sweep_tx.txid.into(),
+        block_hash: sweep_tx.block_hash.into(),
+    };
+    db.write_transaction(&sweep_tx).await.unwrap();
+    db.write_bitcoin_transaction(&bitcoin_tx_ref).await.unwrap();
+
+    // Generate the transaction and corresponding request context.
+    let (mut complete_deposit_tx, ctx) = make_complete_deposit(&deposit_req, &sweep_tx, &chain_tip);
+    // Different: The amount cannot exceed the amount in the deposit
+    // request.
+    complete_deposit_tx.amount = deposit_req.amount + 1000;
```

Comment:
always better to just use `1` in situations like this imo as it proves the logic is sensitive (i.e. in a `>=` vs. `>` scenario)